### ISSUE TITLE
Get IridescentDishWithOlives glTF 2.0 sample model working with X3DOM

### ISSUE THREAD

**Issue by coderextreme** [x3dom/x3dom#1158]
https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/IridescentDishWithOlives

**Comment by cx20** [comment]
BTW, this model has been added to [gltf-test](https://github.com/cx20/gltf-test) and can be tested with `X3DOM`.

https://cx20.github.io/gltf-test/
![image](https://user-images.githubusercontent.com/1833979/126603870-e1aa3f3c-51c8-440a-a20d-5d75e344e06a.png)

[X3DOM v1.8.2 + IridescentDishWithOlives.gltf result:](https://cx20.github.io/gltf-test/examples/x3dom/index.html?category=tutorialModels&model=IridescentDishWithOlives&scale=8&type=glTF)

![image](https://user-images.githubusercontent.com/1833979/126604114-2d1e6765-238a-4cb3-a499-f8ad6501409c.png)

**Comment by andreasplesch** [comment]
Just to note that this (and other new) examples use vendor extensions to gltf 2. Hopefully, some of the extensions can be adopted by gltf.

**Comment by coderextreme** [comment]
As far as I am aware, they are KHR extensions and they are in gltf-test, so
you can see who is adopting and their progress.   Even X3DOM is in
gltf-test.

Perhaps X3D vendors can add extensions to glTF?  PROTOs?

John

**Comment by coderextreme** [comment]
For example, three.js:

https://cx20.github.io/gltf-test/examples/threejs/index.html?category=tutorialModels&model=IridescentDishWithOlives&scale=8&type=glTF

John

**Comment by coderextreme** [comment]
But yeah, there could be patents, it seems.

**Comment by coderextreme** [comment]
See ratified extensions:
https://github.com/KhronosGroup/glTF/blob/master/extensions/README.md

Enjoy!

John

**Comment by cx20** [comment]
> Just to note that this (and other new) examples use vendor extensions to gltf 2. Hopefully, some of the extensions can be adopted by gltf.

Yes. There are many `KHR` extensions ([IOR](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_ior), [Specular](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_specular), [Transmission](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_transmission) and [Volume](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_volume)) used in this model.

Since last year, Khronos has been continuously developing specifications for new glTF extensions.
The activity seems to be called [PBR Next](https://github.com/KhronosGroup/glTF/milestone/2).
These extensions focus on the features needed to represent the product in 3D.

I realize that it takes a lot of time and effort to implement a new extension, so I can't ask you to do it lightly, but I would like to see it happen someday.

Dec 2020 - glTF PBR Next phase 1
https://www.khronos.org/news/press/khronos-releases-wave-of-new-gltf-pbr-3d-material-capabilities

|extension name             |
|:--------------------------|
|[KHR_materials_clearcoat](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_clearcoat)    |
|[KHR_materials_transmission](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_transmission) |
|[KHR_materials_sheen](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_sheen)        |


Jul 2021 - glTF PBR Next phase 2
https://www.khronos.org/news/press/new-gltf-extensions-raise-the-bar-on-3d-asset-visual-realism

|extension name             |
|:--------------------------|
|[KHR_materials_ior](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_ior)          |
|[KHR_materials_volume](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_volume)       |
|[KHR_materials_specular](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_specular)     |

**Comment by coderextreme** [comment]
I forgot to mention this is X3DJSON using X3DOM.  gltf-test has much 
better results for X3DOM, and I'd like to get to the bottom of it.  I'm 
guessing that some how, the XML textures were not converted from glTF to 
XML properly, or I have a bad path.

John

**Comment by andreasplesch** [comment]
Thanks for the overview. It sounds like KHR ratified extensions falling under the Khronos IP framework (details?) will be there to stay, possibly even substituting for a core glTF 3.0 release.

**Comment by coderextreme** [comment]
https://www.khronos.org/members/ip-framework

Looks like there's an Adopter's fee as well as a separate Member fee?

Plus patents are not discussed, but there's some protection for 
implementers?

What was required for PBR?

What is the Web3d/Khronos deal?

John

**Comment by coderextreme** [comment]
https://www.khronos.org/conformance/adopters

I'm not seeing a charge for glTF 2.0 (plus extensions)?

So there's probably still some patents??

John

**Comment by coderextreme** [comment]
Okay, I would look up specifics of glTF on the Khronos site and 
Wikipedia.  It appears to be separately treated.

John

**Comment by coderextreme** [comment]
I'm thinking there's separate licensing between glTF and glTF extensions?

Clarification?

John

**Comment by cx20** [comment]
I'm not a Khronos Group person, so I don't know the exact details, but I don't think there are any patent royalties involved in developing the glTF extension.

@emackey This may be a rudimentary question, but let us know.
Do we need some kind of agreement with Khronos to implement the glTF PBR extension to glTF in library?

**Comment by emackey** [comment]
Disclaimer, I'm not a lawyer, and this post does not constitute legal advice.

Khronos glTF specifications, as well as *ratified* `KHR_` glTF extensions, are completely royalty-free and patent-free/granted to the best of Khronos' ability to ensure it.  One of the tools used to ensure this is the IP Framework, linked earlier in this thread.

All of the glTF extensions listed in the [Ratified glTF Extensions Section](https://github.com/KhronosGroup/glTF/tree/master/extensions#ratified-khronos-extensions), including all the extensions used in IridescentDishWithOlives, have been through the Khronos ratification cycle.  This is a 6-week cycle where all of the Khronos member organizations have been given notice of the extension being ratified, and asked to disclose any IP they have (including patents, copyrights, any form of Intellectual Property) over the proposed extension.  If no claims or objections are heard during the ratification review period, the proposed extension is sent to the Khronos Board of Promoters (representatives from top-level Khronos member companies) for a formal vote.

When and if ratification succeeds, as it has already done for the above extensions, none of the Khronos member companies may attempt to charge licensing fees, patent fees, or any other fees outside of Khronos, even if they do hold patents or other IP on the technology, because of the IP Framework they agreed to when they joined Khronos.

It is possible, however unlikely, that some external company, that has never been a Khronos member, could attempt to charge a patent fee or something without ever signing the IP Framework.  Khronos certainly takes this risk very seriously, and takes additional protective steps against this scenario.  To the best of my knowledge this has not yet happened to glTF or any of its extensions.

The Vendor and Multi-Vendor (`EXT_`, etc) extensions have never been through Khronos ratification, and may have their own licensing arrangements per extension.  Generally all the ones listed on the glTF GitHub page are believed to be free, but without the rigorous IP protection of the ratified `KHR_` ones.

The "Adopters" agreements and fees are for OpenGL (folks building OpenGL drivers for gfx cards).  They are not applicable to anything related to glTF.

**TL;DR**: Yes, glTF and its ratified `KHR` extensions are completely free for everyone to use, always.

In addition, this particular sample model happens to be free under a CC-BY 4.0 license from Wayfair.  The glTF Sample Viewer implementation of the new extensions is free under an Apache 2.0 license.

**Comment by cx20** [comment]
@emackey Thank you for your accurate advice.
